src/expr/node_manager_template.cpp

```diff
@@ -941,6 +958,39 @@ TypeNode NodeManager::mkTupleType(const std::vector<TypeNode>& types)
   return d_tt_cache.getTupleType(this, types);
 }
 
+TypeNode NodeManager::mkNullableType(const TypeNode& type)
+{
+  Assert(!type.isNull());
+  auto it = d_nt_cache.find(type);
+  if (it != d_nt_cache.end())
+  {
+    return it->second;
+  }
+  // construct the corresponding datatype with two constructors
+  // __cvc5_nullable_type_ctor_null, and __cvc5_nullable_type_ctor_some
+  std::stringstream sst;
+  sst << "__cvc5_nullable_" << type;
+  DType dt(sst.str());
+  dt.setNullable();
+  std::stringstream sscNull;
+  sscNull << sst.str() << "_ctor_null";
```

Comment:
You should just be able to give this its proper name, i.e. `nullable.null`.  This should also avoid the need for some of the special casing in the smt2 printer.

Reply:
Done.